File: DSAlgorithms/Leetcode/c++/arrays_hash/49-group_anagrams.cpp
vector<vector<string>> groupAnagrams(vector<string>& strs) {
    std::vector<std::vector<std::string>> ans;
    std::unordered_map<std::string, std::vector<std::string>> hash;
    for(int i = 0; i < strs.size(); i++)    
    {
        std::string current_string = strs[i];
        std::sort(strs[i].begin(), strs[i].end());
        hash[strs[i]].push_back(current_string);
    }
    for(auto i: hash)
    {
        ans.push_back(i.second);
    }
    return ans;
    }

File: DSAlgorithms/Leetcode/ts/src/arrays_hash/49-group_anagrams.ts
const groupAnagrams = (strs: string[]): string[][] =>{
    const result: string[][] = [];
    const hash = new Map<string, string[]>();
    for(let i = 0; i < strs.length; i++)
    {
        let current = strs[i];
        let values: string[] = [];
        let sorted: string = strs[i].split('').sort().join('');
        let in_hash = hash.has(sorted);
        if(in_hash)
            values = hash.get(sorted);
        if(!in_hash)
            values = [];
        values.push(current);
        hash.set(sorted, values);
    }
    hash.forEach((value, key)=>{
        result.push(value);
    })
    return result;
};
